internal/expr/prepare.go

```diff
@@ -0,0 +1,73 @@
+package expr
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// PreparedExpr contains an SQL expression that is ready for execution.
+type PreparedExpr struct {
+	ParsedExpr *ParsedExpr
+	SQL        string
```

Comment:
I would make those details private, as one of them is the input to the query, and the other one is the actual "prepared" expression. If we expose it, we'll lose the opportunity to do anything else with this type without breaking its promises.

Reply:
I've changed `ParsedExpr` to `inputs` and made it private. I've left SQL as public because we use it in testing and it makes sense that the `PreparedExpr` should have the prepared SQL.